### PR TITLE
[Ven-1559] [VEN-1560]: require initial liquidity adding a market and avoid seizing on markets disabled as collateral

### PR DESCRIPTION
**Pull request by defcon022**
## Description

Resolves #
- VEN-1559
- VEN-1560

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Pool/PoolRegistry.sol [hunk 1]
```diff
@@ -276,6 +276,8 @@ contract PoolRegistry is Ownable2StepUpgradeable, AccessControlledV8, PoolRegist
             "PoolRegistry: Market already added for asset comptroller combination"
         );
 
+        require(input.initialSupply > 0, "PoolRegistry: initialSupply is zero");
```
Comment (defcon022) [inline contracts/Pool/PoolRegistry.sol:279]: I could have used custom error here but require is used in other places in the contract, to maintain consistency I have used require.

File: contracts/Comptroller.sol [hunk 2]
```diff
@@ -525,6 +530,10 @@ contract Comptroller is
             }
         }
 
+        if (!market.accountMembership[borrower]) {
+            revert MarketNotCollateral(vTokenCollateral, borrower);
```
Comment (defcon022) [inline contracts/Comptroller.sol:534]: Will add fork tests for this case as UT does not exist for this method.

File: tests/hardhat/PoolRegistry.ts [hunk 3]
```diff
@@ -350,6 +350,12 @@ describe("PoolRegistry: Tests", function () {
       ).to.be.revertedWithCustomError(poolRegistry, "ZeroAddressNotAllowed");
     });
 
+    it("reverts if vTokenReceiver address is zero", async () => {
```
Comment (chechu) [inline tests/hardhat/PoolRegistry.ts:353]:
```suggestion
    it("reverts if initial supply is zero", async () => {
```

Comment (chechu) [inline tests/hardhat/PoolRegistry.ts:353]:
```suggestion
    it("reverts if initial supply is zero", async () => {
```

Comment (defcon022) [reply tests/hardhat/PoolRegistry.ts]: Fixed :sweat_smile: 7b9208c5ce0b946776fc4828cbb1e852514fb29e

**Review by chechu — APPROVED** [review]

File: contracts/Comptroller.sol (hunk 2 above)
Comment (defcon022) [reply contracts/Comptroller.sol:534]: Added the UT

Comment (defcon022) [reply contracts/Comptroller.sol:534]: Added the UT

**Pull request merged**